lib/gollum-lib/wiki.rb

```diff
@@ -810,19 +809,18 @@ def inspect
 
     private
 
-    # Prefixes page_file_dir to the given directory
+    # Conjoins elements of a page or file path and prefixes the page_file_dir.
+    # Throws Gollum::IllegalDirectoryPath if page_file_dir is set, and the resulting
+    # path is not below it (e.g. if the dir or name contained '../')
     #
-    # dir - The String directory path to be sanitized
+    # dir    - The String directory path
+    # name   - The String name of the Page or File
+    # format - The Symbol format of the page. Should be nil for Files.
     #
-    # Returns a String path
-    def sanitized_dir(dir)
-      sanitized  = ::File.join([@page_file_dir, '/', dir].compact)
-      raise Gollum::IllegalDirectoryPath if @page_file_dir && !Pathname.new(sanitized).cleanpath.to_s.start_with?(@page_file_dir)
-      sanitized
-    end
-
-    def merge_page_elements(dir, name, format)
-      result = ::File.join(*[sanitized_dir(dir), self.page_file_name(name, format)])
+    # Returns a String path.  .
+    def merge_path_elements(dir, name, format)
+      result = ::File.join([@page_file_dir, '/', dir, self.page_file_name(name, format)].compact)
+      raise Gollum::IllegalDirectoryPath if @page_file_dir && !Pathname.new(result).cleanpath.to_s.start_with?(@page_file_dir)
```

Comment:
It feels a bit weird to throw this error in this helper method. What would be a better place, you think? In `write_page_or_file`? Downside of that is other methods, like `delete_file`, wouldn't benefit from the protection against directory traversal. So maybe in `Committer#add_to_index`? That also seems strange..